src/appshell/commandlinecontroller.cpp

```diff
@@ -40,18 +41,35 @@ void CommandLineController::parse(const QStringList& args)
 
 void CommandLineController::apply()
 {
-    // Common
-    if (m_parser.isSet("D")) {
+    auto floatValue = [this](const QString& name) -> std::optional<float> {
         bool ok;
-        float val = m_parser.value("D").toFloat(&ok);
+        float val = m_parser.value(name).toFloat(&ok);
         if (ok) {
+            return val;
+        } else {
+            return std::nullopt;
+        }
+    };
+
+    // Common
+    if (m_parser.isSet("D")) {
```

Comment:
Maybe better to make named constants for all options? Like this:

static const QString CHANGE_DPI_OPTION("D")

Reply:
I think it will not be convenient.
Looking for where the option is processed ...
If you make constants, you must first find the name of the constant, and then the place where it is processed.